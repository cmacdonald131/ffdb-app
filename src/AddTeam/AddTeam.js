import React, { Component } from 'react';
import './AddTeam.css';



class AddTeam extends Component {
    render() {
        return (
            <div className="AddTeam">
                <header>
                    <h1>Add your team</h1>
                </header>
                <main className="AddTeamSection">
                    <form className="add-team">
                        <section className="form-section">
                            <label>Team Name</label>
                            <input type="text" name="team-name" placeholder="Team Name" required></input>
                        </section>
                        <section className="form-section">
                            <label>Website</label>
                            <input type="text" name="website" placeholder="Enter your team site" required></input>
                        </section>
                        <section className="form-section">
                            <label>Username</label>
                            <input type="text" name="username" placeholder="Enter username" required></input>
                        </section>
                        <section className="form-section">
                            <label>Password</label>
                            <input type="text" name="password" placeholder="Enter password" required></input>
                        </section>
                        <button type="submit">Submit</button>
                        <button type="reset">Reset</button>
                        <button type="cancel">Cancel</button>
                    </form>
                </main>
            </div>
        );
    }
}
export default AddTeam;
