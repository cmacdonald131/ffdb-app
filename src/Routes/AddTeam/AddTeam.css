* {
  box-sizing: border-box;
}

.AddTeam {
  background: url(../../Images/football_field3.jpg);
  background-size: cover;
  background-position: center center;
  min-height: 110vh;
}

.label {
  display: block;
}

form {
  text-align: left;
}

input {
  width: 100%;
  align-items: center;
}

.AddTeamSection {
  align-content: center;
  margin: 20px;
  padding: 10px 10px;
}

h1 {
  color: #ffffff;
  margin-top: 20px;
}

button {
  order: 1px solid #aaa;
  background-color: #eee;
  font-size: 1.1em;
  border-radius: 4px;
  padding: 8px 12px;
  font-weight: bold;
  cursor: pointer;
  text-align: center;
}

.Submit {
  margin-right: 5px;
}

.cancel {
  margin-left: 5px;
}

@media screen and (min-width: 700px) {
  .label {
    margin-left: 130px;
  }

  .Submit {
      margin-left: 30%;
      margin-right: 10px;
  }

  .cancel {
      margin-right: 30%;
      margin-left: 10px;
  }
}

@media screen and (min-width: 1100px) {
    .Submit {
        margin-right: 20px;
        margin-left: 39%;
    }

    .cancel {
        margin-left: 20px;
    }
}