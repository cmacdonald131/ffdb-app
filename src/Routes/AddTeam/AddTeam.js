import React, { Component } from 'react'
import config from '../../config'
import './AddTeam.css'
import Navbar from '../../Navbar/Navbar'
import ApiContext from '../../ApiContext'
import TokenService from '../../Services/token-service'
import { Input, Required } from '../../Utils/Utils'
import uuid from 'uuid/v4'


class AddTeam extends Component {
    static contextType = ApiContext
    onSubmit = (e) => {
        e.preventDefault();
        const teamname = e.target.teamname.value;
        const website = e.target.website.value;
        const username = e.target.username.value;
        const password = e.target.password.value;
        const id = uuid;

        fetch(`${config.API_ENDPOINT}/teams`, {
            method: "Post",
            body: JSON.stringify({
                name: teamname,
                website: website,
                username: username,
                password: password,
                id: id,

            }),
            headers: {
                'Authorization': `Bearer ${TokenService.getAuthToken()}`,
                'content-type': 'application/json'
            }
        })

            .then(team => {
                return team.json()
            }).then(data => {
                this.props.history.push('/team-page')
            })

    }

    onCancel = (e) => {
        e.preventDefault();
        this.props.history.push('/team-page')
    }

    render() {
        return (
            <div className="AddTeam">
                <Navbar />
                <header>
                    <h1>Add your team</h1>
                </header>
                <main className="AddTeamSection">
                    <form onSubmit={this.onSubmit} className="add-team">
                        <section className="form-section">
                            <label htmlFor="teamname" className="label"
                            >Team Name <Required />
                            </label>
                            <Input
                                type="text"
                                name="teamname"
                                placeholder="Team Name">
                            </Input>
                        </section>
                        <section className="form-section">
                            <label htmlFor="website" className="label"
                            >Website <Required />
                            </label>
                            <Input
                                type="text"
                                name="website"
                                placeholder="Enter your team site">
                            </Input>
                        </section>
                        <section className="form-section">
                            <label htmlFor="username" className="label"
                            >Username <Required />
                            </label>
                            <Input
                                type="text"
                                name="username"
                                placeholder="Enter username">
                            </Input>
                        </section>
                        <section className="form-section">
                            <label htmlFor="password" className="label"
                            >Password <Required />
                            </label>
                            <Input
                                type="text"
                                name="password"
                                placeholder="Enter password">
                            </Input>
                        </section>
                        <button type="submit" className="Submit">Submit</button>
                        <button type="reset" className="reset">Reset</button>
                        <button onClick={this.onCancel} type="cancel" className="cancel">Cancel</button>
                    </form>
                </main>
            </div>
        );
    }
}
export default AddTeam;
