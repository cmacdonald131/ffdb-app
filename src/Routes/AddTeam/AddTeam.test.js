import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddTeam from './AddTeam'

describe(`AddTeam component`, () => {
    const props = {
        teamname: 'test-teamname',
        username: 'test-username',
        password: 'test-password',
        website: 'test-website'

    }

    it('renders a .AddTeam by default', () => {
        const wrapper = shallow(<AddTeam />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the AddTeam given props', () => {
        const wrapper = shallow(<AddTeam {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})