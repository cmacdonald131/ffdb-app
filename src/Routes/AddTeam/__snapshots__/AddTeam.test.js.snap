// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTeam component renders a .AddTeam by default 1`] = `
<div
  className="AddTeam"
>
  <Navbar />
  <header>
    <h1>
      Add your team
    </h1>
  </header>
  <main
    className="AddTeamSection"
  >
    <form
      className="add-team"
      onSubmit={[Function]}
    >
      <section
        className="form-section"
      >
        <label
          className="label"
          htmlFor="teamname"
        >
          Team Name 
          <Required />
        </label>
        <Input
          name="teamname"
          placeholder="Team Name"
          type="text"
        />
      </section>
      <section
        className="form-section"
      >
        <label
          className="label"
          htmlFor="website"
        >
          Website 
          <Required />
        </label>
        <Input
          name="website"
          placeholder="Enter your team site"
          type="text"
        />
      </section>
      <section
        className="form-section"
      >
        <label
          className="label"
          htmlFor="username"
        >
          Username 
          <Required />
        </label>
        <Input
          name="username"
          placeholder="Enter username"
          type="text"
        />
      </section>
      <section
        className="form-section"
      >
        <label
          className="label"
          htmlFor="password"
        >
          Password 
          <Required />
        </label>
        <Input
          name="password"
          placeholder="Enter password"
          type="text"
        />
      </section>
      <button
        className="Submit"
        type="submit"
      >
        Submit
      </button>
      <button
        className="reset"
        type="reset"
      >
        Reset
      </button>
      <button
        className="cancel"
        onClick={[Function]}
        type="cancel"
      >
        Cancel
      </button>
    </form>
  </main>
</div>
`;

exports[`AddTeam component renders the AddTeam given props 1`] = `
<div
  className="AddTeam"
>
  <Navbar />
  <header>
    <h1>
      Add your team
    </h1>
  </header>
  <main
    className="AddTeamSection"
  >
    <form
      className="add-team"
      onSubmit={[Function]}
    >
      <section
        className="form-section"
      >
        <label
          className="label"
          htmlFor="teamname"
        >
          Team Name 
          <Required />
        </label>
        <Input
          name="teamname"
          placeholder="Team Name"
          type="text"
        />
      </section>
      <section
        className="form-section"
      >
        <label
          className="label"
          htmlFor="website"
        >
          Website 
          <Required />
        </label>
        <Input
          name="website"
          placeholder="Enter your team site"
          type="text"
        />
      </section>
      <section
        className="form-section"
      >
        <label
          className="label"
          htmlFor="username"
        >
          Username 
          <Required />
        </label>
        <Input
          name="username"
          placeholder="Enter username"
          type="text"
        />
      </section>
      <section
        className="form-section"
      >
        <label
          className="label"
          htmlFor="password"
        >
          Password 
          <Required />
        </label>
        <Input
          name="password"
          placeholder="Enter password"
          type="text"
        />
      </section>
      <button
        className="Submit"
        type="submit"
      >
        Submit
      </button>
      <button
        className="reset"
        type="reset"
      >
        Reset
      </button>
      <button
        className="cancel"
        onClick={[Function]}
        type="cancel"
      >
        Cancel
      </button>
    </form>
  </main>
</div>
`;
